scripts/release.py

```diff
@@ -280,7 +319,9 @@ def main(args: argparse.Namespace) -> None:
     write_version_file(version)
     write_version_to_pyproject(version)
 
-    generate_changelog(version)
+    if not version.prerelease:
+        # never update changelog on an alpha version
+        generate_changelog(version)
     base = git_current_branch()
     branch = create_release_branch(version)
```

Comment:
I think for alpha releases we need to differentiate: if the current branch is neither master nor a release branch (matching `1.2.x` pattern) I think we should directly commit to the current branch. this seems to work better for the workflow where you e.g. break model compatibility on a feature branch and need to increase the version number.

Reply:
OK gotcha